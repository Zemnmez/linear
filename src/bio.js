import bio from '@zemnmez/bio';

const parseDates = (data) => {
  data.timeline = data.timeline.map(({date, ...etc}) => {
    date = parseSimpleDate(date);

    return ({date, ...etc});
  });

  return data;
}

const months = "Jan Feb Mar Apr May Jun Jul Aug Sep Oct Nov Dec".split(" ");
const parseSimpleDate = (date) => {
    let [month, day, year] = date.split(" ");
    month = months.indexOf(month);

    if (month === -1) throw Error(`invalid date ${date}`);

    return new Date(year, month, day);
}

const newBio = parseDates(bio);

export default newBio;
