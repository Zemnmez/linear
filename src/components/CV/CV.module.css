@media print {
  .CV { border: 0 }
}

.cv {
  font-size: 80%;
  box-sizing: border-box;
  padding: 20mm 30mm;
  overflow: hidden;
  width: 210mm;
  height: 297mm;
  margin: auto ;
  border: 1px solid var(--fgc);

  display: grid;
  grid: "header header header header header" min-content
        "experience-title experience-title experience-title experience-title experience-title" min-content
        ". . . . ." .4fr
        "experience experience experience experience experience" 7fr
        "works-title works-title works-title works-title works-title" min-content
        ". . . . ." .4fr
        "works works works works works" 1fr
        ". . . . ." .4fr
        "works2 works2 works2 works2 works2" 1fr
        ". . . . ." .4fr
        "skills-title skills-title skills-title skills-title skills-title" min-content
        ". . . . ." .4fr
        "skills skills skills skills skills" 1fr
        ". . . . ." 3em
        ". dash-left future dash-right ." 2em / 1fr 2em 1fr 2em 1fr;
        /* yeah i know this is kinda disgusting and i should make a dashed Future component... */
}

.cv::before { content:"–"; grid-area:dash-left; margin:auto; opacity: .5 }
.cv::after { content:"–"; grid-area: dash-right; margin: auto; opacity: .5 }

.email { grid-area: email }
.phone { grid-area: phone }
.name { grid-area: name }
.date { grid-area: date }

.header {
  grid-area: header;
  display: grid;
  justify-items: center;
  align-items: center;
  grid: "website icon date" min-content
        ". . ." 1em
        "phone name email" min-content
        ". . ." 2em
        / 1fr 1fr 1fr;
}

.header .website { grid-area: website }
.header .headerIcon { grid-area: icon ; height: 2.5em }

.experience {
  grid-area: experience;
  display: grid
}

.work {
  display: grid;
  grid: "employer employer . start dash end" min-content
        "position position . duration duration duration" min-content
        ". . . . . ." .1em
        ". content content . . ." min-content
        ". . . . . ." 1em
        / 1em 1fr 1em 4.5em 2em 4em;

}

.work p { margin: 0 }
.work p + p { margin-top: 1em }

.work::before {
  content: " - ";
  grid-area: dash;
  justify-self: center;
}

.work .start,
.work .end,
.work .duration { white-space: nowrap; opacity: .6 }
.work .duration { opacity: .4 }

.work .employer { grid-area: employer ; font-weight: bold; font-style: italic }
.work .position { grid-area: position }
.work .start { grid-area: start; justify-self: end }
.work .end { grid-area: end ; justify-self: end }
.work .duration { grid-area: duration ; justify-self: end }
.work .content { grid-area: content }
.work .content p:last-of-type::after { content: "." }
.work .content:first-letter, .work .content *:first-letter { text-transform: uppercase }
.work .content ul { list-style-type: none }
.work .content ul { margin: 0 ; padding: 0 }
.work .content li + li { margin-top: 1em }

.skills, .works {
  grid-area: skills;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(7em, 1fr));
  grid-auto-flow: dense;
  align-items: center;
  justify-items: stretch;
  text-align: center;
}


.worksTitle { grid-area: works-title }
.experienceTitle { grid-area: experience-title }

.cv a:link { text-decoration: none }

.works { grid-area: works; }
.works2 { grid-area: works2; }

.future { grid-area: future; stroke: none; fill: var(--fgc) !important }

.skills.works { grid-gap: 0 1em }

.skillsTitle { grid-area: skills-title }
.skills:not(.works) .description { display: none }

.skills .description {
  box-sizing: border-box;
  opacity: .7;
  grid-row: 2;
  align-self: start;
}

.skills :not(.description) {
  text-transform: lowercase
}
