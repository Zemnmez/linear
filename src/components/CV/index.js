import React from 'react';
import { Name } from 'components/Profile';
import Moment from 'react-moment';
import Icon from 'components/SadHumans';
import Rule from 'components/Rule';
import Future from 'components/Future';
import { Description } from 'components/Timeline';
import bcrypt_whitelist from 'knowitwhenyouseeit';
import style from './CV.module.css';

const hurl = (message) => { throw new Error(message) }

export default ({
  data: {
    who: {name: names, handle},
    bio,
    timeline,
    skills = []
  },
  className,
  focuses = "",
  limit = 6,
  phone = "",
  email = ""
}) => {

    let of_note = timeline.filter(({ priority }) => priority >=6)
        .filter(({ tags, description }) =>
          tags.some(tag => tag === "security") && !tags.some(tag => tag === "work")
        ).sort(({ priority: a}, {priority: b}) => b - a)

    const [of_note_1, of_note_2] = [of_note.slice(0,4), of_note.slice(4,8)]


  return <div {...{
    className: [style.cv].concat(className).join(" ")
  }}>

    <Header {...{
      email: bcrypt_whitelist("$2a$11$4ad28pmQZw7z1Mi.DheOBeCgo2qa323.yzcxWUNTA6Y1Q4GfzqiVC")(email),
      phone: bcrypt_whitelist("$2a$11$SQtYc3STHgrijkuoDiO2O.X020hozBXl06lam/DIEdyMiGUKoPcWm")(phone),
      names
    }} />


    <Rule className={style.experienceTitle}>selected experience</Rule>

    <Experience {...{
      events: timeline.filter(({ tags }) => tags.some(tag => tag === "work"))
        .sort(({ date: a }, { date: b }) => b - a).slice(0, 4)
    }} />

    <Rule className={style.worksTitle}>of note</Rule>

    <Skills {...{
      skills: skills.sort((a,b) => a > b? 1 : -1).slice(0, 12).map(skill => ({title: skill}))
    }} />

    <Rule className={style.skillsTitle}>skills</Rule>

    <Skills className={style.works} {...{
      skills: of_note_1
    }} />

    <Skills className={style.works2} {...{
      skills: of_note_2
    }} />


    <Future className={style.future}/>

  </div>
}

const Skills = ({ skills, className }) => <div {...{
    className: [style.skills].concat(className).join(" ")
  }}>

  {skills.map((skill, i) => <Skill {...{...skill, key: skill.title}} />)}
</div>

const Skill = ({ title, description, className }) => [
  <div className={style.title}>{title}</div>,
  <div className={style.description}>{description}</div>
]

const t = { milliseconds: 1 };
t.seconds = t.milliseconds * 1000;
t.minutes = t.seconds * 60;
t.hours = t.minutes * 60;
t.days = t.hours * 24;
t.weeks = t.days * 7;
t.years = t.days * 365;

let parseDuration = (duration) => {
  const [n, unit] = duration.split(" ");
  return (+n) * (t[unit] || hurl(`unknown unit ${unit}`));
}

/*const roundUpToNearestYear = (date) =>
  new Date(date.getFullYear(),12); */

const roundDownToNearestYear = (date) =>
  new Date(date.getFullYear()-1,12)

const Experience = ({ events, className }) => <div {...{
    className: [style.experience].concat(className).join(" ")
  }}>

  {events.map((event, i) => <Work key={i} {...event} />)}
</div>

const aToOne = (str) => str.replace(/\ba\b/g, "1");
const Work = ({ date, description, duration, title, points, className }) => {

  // rough
  const [position, employer] = title.split(",").map(v => v.trim());

  duration = duration === "ongoing"? undefined: parseDuration(duration);

  return <div {...{
      className: [style.work].concat(className).join(" ")
    }}>

  <div className={style.employer}>{employer}</div>
  <div className={style.position}>{position}</div>
  <When {...{date: date, className: style.start}} />
  {duration?
    <When {...{date: new Date( (+date) + duration), className: style.end}}/>:
    <div className={style.end}>Present</div>
  }

  {duration?
      <div className={style.duration}><Moment filter={aToOne} ago to={roundDownToNearestYear(new Date( (+date) + duration))}>{date}</Moment></div>:
      <div className={style.duration}><Moment filter={aToOne} ago toNow>{roundDownToNearestYear(date)}</Moment></div>
  }

  <Description {...{
    className: style.content,
    description
  }} />

  </div>
}


const Header = ({ email, phone, names, className }) => <div {...{
    className: [style.header].concat(className).join(" ")
  }}>

    <a className={style.website} href={process.env.PUBLIC_URL}>zemn.me</a>
    <Email {...{email}} />
    <Phone {...{phone}}/>
    <Icon {...{className: style.headerIcon}} />
    <When {...{format: "MMM YYYY"}} />
    <Name {...{className: style.name, names}}/>
</div>




const Email = ({ className, email }) => <div {...{
    className: [style.email].concat(className).join(" ")
  }}>

  {email}
</div>

const Phone = ({ className, phone }) => <div {...{
    className: [style.phone].concat(className).join(" ")
  }}>

  {phone}
</div>

const When = ({ className, format = "YYYY", date = new Date() }) => <div {...{
    className: [style.date].concat(className).join(" ")
  }}>

  <Moment {...{format}}>{date}</Moment>
</div>
