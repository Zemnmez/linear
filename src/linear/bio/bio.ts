import { SimpleDate, Parse as ParseSimpleDate } from 'linear/time/simpledate';
import { Upgrade } from 'linear/util';
import { KV } from 'linear/higher';

export interface BioJson {
    bio: string,
    birthdate: SimpleDate,
    employment: EmploymentJson[],
    skills: string[],
    links: Record<string, string>,
    timeline: JsonEvent[]
    who: Who
}

export type Bio = Upgrade<BioJson, {
    birthdate: Date,
    timeline: Timeline,
    links: KV<string, URL>,
    employment: Employment[]
}>

export type Timeline = Event[];

export interface Who {
    handle: string,
    name: string[]
}


export interface EmploymentJson {
    since: SimpleDate,
    title: string,
    where: string
}

export type Employment = Upgrade<EmploymentJson,{
    since: Date
}>

export const Employment:
    (v: EmploymentJson) => Employment
=
    ({ since, ...etc }) => ({
        since: ParseSimpleDate(since),
        ...etc
    })
;

export interface JsonEvent {
    date: SimpleDate,
    description?: string,
    tags?: string[],
    title: string,
    url?: string,
    priority?: number,
    duration?: string | "ongoing"
}

export type Event = Upgrade<JsonEvent,{
    date: Date,
    url?: URL
}>


export const ParseBio:
    (b: BioJson) => Bio
=
    ({birthdate, timeline, links, employment, ...etc}) => ({
        birthdate: ParseSimpleDate(birthdate),
        timeline: Timeline(timeline),
        employment: employment.map(Employment),
        links: new Map(Object.entries(links).map(([k,v]) =>
            [k, new URL(v)])),
        ...etc
    })
;

export const Timeline:
    (t: JsonEvent[]) => Event[]
=
    t => t.map(Event).sort(
        ({ date: A }, { date: B }) => (+A) - (+B)
    )
;

const Event:
    (e: JsonEvent) => Event
=
    ({ date, url, ...etc }) => ({
        date: ParseSimpleDate(date),
        url: url === undefined? url: new URL(url),
        ...etc
    })
;

export const Bio = ParseBio({
    "bio": "security engineer with strong skills in the Go programming language and the design of robust modern application security architectures",
    "birthdate": [17, "may", 1994],
    "employment": [
        {
            "since": ['sep', 2014],
            "title": "Information Security Engineer",
            "where": "Twitch"
        },
        {
            "since": ['may', 2018],
            "title": "Application Security Engineer",
            "where": "UKNCSC"
        }
    ],
    "links": {
        "linkedin": "https://www.linkedin.com/in/thomas-shadwell-4b333b50",
        "github": "https://github.com/zemnmez",
        "twitch": "https://twitch.tv/zemnmez",
        "twitter": "https://twitter.com/zemnmez"
    },
    "skills": [
        "Go",
        "Python",
        "Javascript",
        "React",
        "d3",
        "Bash",
        "Ruby",
        "appsec arch",
        "electron",
        "AWS",
        "pentesting",
        "code review",
        "cryptography",
        "reversing"
    ],
    "timeline": [
        {
            date: [11, 'jul', 2019],
            title: "National Cyber Security Centre 'Turing' challenge coin",
            priority: 5,
            tags: [ 'accolade' ],
            url: "https://twitter.com/zemnmez/status/1149278890969456640",
            description: "award for my work on UK government vulnerability disclosure policy and my responsible disclosure of vulnerabilities in the UK tax system."
        },
        {
            "date": [20, 'jan', 2019],
            "description": "hybrid ctf / esports competition winners",
            "priority": 6,
            "tags": [
                "accolade",
                "security",
                "gaming"
            ],
            "title": "hack fortress 2019 champions",
            "url": "https://twitter.com/tf2shmoo/status/1086785642514796544"
        },
        {
            "date": [14, 'jan', 2019],
            "description": "react helper bindings for d3",
            "priority": 6,
            "tags": [
                "software",
                "library",
                "d3",
                "react"
            ],
            "title": "reactive-d3",
            "url": "https://github.com/Zemnmez/reactive-d3"
        },
        {
            "date": [7, 'jan', 2019],
            "description": "react based personal website for 2019",
            "priority": 6,
            "tags": [
                "software",
                "react"
            ],
            "title": "linear",
            "url": "https://github.com/Zemnmez/linear"
        },
        {
            "date": [8, 'jan', 2019],
            "description": "news coverage of steam rce",
            "priority": 7,
            "tags": [
                "security",
                "disclosure",
                "comment"
            ],
            "title": "$7,500 Steam Weakness Let Hackers Take Remote Control Of Gamers' PCs",
            "url": "https://www.forbes.com/sites/thomasbrewster/2019/01/08/7500-steam-weakness-let-hackers-take-remote-control-of-gamers-pcs"
        },
        {
            "date": [12, 'aug', 2019],
            "priority": 6,
            "tags": [
                "security",
                "gaming",
                "accolade"
            ],
            "title": "hack fortress DEFCON 2018 winners",
            "url": "https://twitter.com/tf2shmoo/status/1028462663368507392"
        },
        {
            "date": [7, 'jan', 2019],
            "description": "vulnerability to remotely access Steam users' computers",
            "priority": 8,
            "tags": [
                "gaming",
                "security",
                "accolade"
            ],
            "title": "Steam Remote Code Execution",
            "url": "https://hackerone.com/reports/409850"
        },
        {
            "date": [15, 'dec', 2018],
            "description": "Quick article on the security of modern desktop web applications",
            "priority": 7,
            "tags": [
                "security"
            ],
            "title": "\u00dcbersicht Remote Code Execution, Spotify takeover",
            "url": "https://medium.com/@Zemnmez/%C3%BCbersicht-remote-code-execution-spotify-takeover-a5f6fd6809d0"
        },
        {
            "date": [22, 'jan', 2018],
            "priority": 8,
            "tags": [
                "gaming",
                "security",
                "accolade"
            ],
            "title": "forbes 30 under 30, tech",
            "url": "https://www.forbes.com/profile/thomas-shadwell",
            "description": "for my work at Twitch, and on responsible disclosure"
        },
        {
            "date": [22, 'jan', 2018],
            "description": "advisory position. Provided expertise to UK cyber advisory / defence group on Go and building security analysis systems",
            "duration": "ongoing",
            "priority": 8,
            "tags": [
                "security",
                "work"
            ],
            "title": "Application Security Engineer, UK National Cyber Security Centre"
        },
        {
            "date": [23, 'nov', 2017],
            "description": "talk at owasp about critical uk tax system flaw in obfuscated system and the 57 day trek to get it fixed",
            "priority": 6,
            "tags": [
                "security",
                "talk",
                "writing"
            ],
            "title": "how to hack the uk tax system: the talk",
            "url": "https://twitter.com/zemnmez/status/933847040198574080"
        },
        {
            "date": [8, 'sep', 2017],
            "description": "news post on manipulation of UK tax data",
            "priority": 6,
            "tags": [
                "security",
                "disclosure",
                "comment"
            ],
            "title": "'Serious' security flaws found on official UK tax site",
            "url": "http://www.bbc.co.uk/news/technology-41188008"
        },
        {
            "date": [25, 'jan', 2016],
            "description": "unauthorized remote shutdown of Buffalo-made network attached storage devices",
            "priority": 6,
            "tags": [
                "security",
                "writing",
                "disclosure"
            ],
            "title": "Buffalo NAS Remote Shutdown",
            "url": "https://packetstormsecurity.com/files/135368"
        },
        {
            "date": [1, 'may', 2012],
            "description": "full stack freelance work building MVPs for London startups and wrangling data for hackathons",
            "duration": "2 years",
            "priority": 7,
            "tags": [
                "work"
            ],
            "title": "Software Engineer, Consultant"
        },
        {
            "date": [16, 'may', 2016],
            "description": "code execution in official Mr Robot site",
            "priority": 7,
            "tags": [
                "security",
                "disclosure",
                "comment"
            ],
            "title": "'Mr. Robot' Web Weaknesses Left Fans And USA Network Vulnerable, Warns Non-Fictional Hacker",
            "url": "https://www.forbes.com/sites/thomasbrewster/2016/05/16/mr-robot-imagetragick-usa-network-wide-open-to-hackers/#7d49f6f66d77"
        },
        {
            "date": [11, 'may', 2016],
            "description": "XSS in Mr Robot official site",
            "priority": 6,
            "tags": [
                "security",
                "disclosure",
                "comment"
            ],
            "title": "Irony Alert: Hacker Finds Vulnerability In Mr Robot Website",
            "url": "https://www.forbes.com/sites/thomasbrewster/2016/05/11/flaw-in-mr-robot-website-allowed-facebook-attack/#747437ef6bed"
        },
        {
            "date": [8, 'sep', 2017],
            "description": "vulnerability allowing manipulation of UK tax system",
            "priority": 8,
            "tags": [
                "security",
                "writing",
                "disclosure"
            ],
            "title": "how to hack the uk tax system, i guess",
            "url": "https://medium.com/@Zemnmez/how-to-hack-the-uk-tax-system-i-guess-3e84b70f8b"
        },
        {
            "date": [18, 'feb', 2017],
            "description": "musings on the evolution of design",
            "tags": [
                "writing"
            ],
            "title": "Design Evolves By Constraint",
            "url": "https://medium.com/@Zemnmez/design-evolves-by-constraint-f2d87697d25e"
        },
        {
            "date": [3, 'jan', 2016],
            "description": "minimal reactive d3.js resistor colour code calculator",
            "priority": 6,
            "tags": [
                "software"
            ],
            "title": "r.no.ms",
            "url": "http://r.no.ms"
        },
        {
            "date": [27, 'apr', 2016],
            "description": "padding oracle based decryption of Steam traffic",
            "priority": 7,
            "tags": [
                "security",
                "gaming",
                "disclosure"
            ],
            "title": "steam patches broken crypto in wake of replay, padding oracle attacks",
            "url": "https://threatpost.com/steam-patches-broken-crypto-in-wake-of-replay-padding-oracle-attacks/117691/"
        },
        {
            "date": [7, 'jul', 2014],
            "description": "exploit using content security policy 1 to steal data on the web",
            "priority": 9,
            "tags": [
                "security",
                "writing",
                "disclosure"
            ],
            "title": "when security creates insecurity",
            "url": "http://archive.is/UXD8j"
        },
        {
            "date": [4, "jul", 2017],
            "description": "musings on go-specific security gotchas",
            "priority": 5,
            "tags": [
                "security",
                "talk"
            ],
            "title": "This Will Cut You: Go's Sharper Edges",
            "url": "https://www.infoq.com/presentations/go-security"
        },
        {
            "date": [1, 'sep', 2011],
            "description": "Volunteer role at once largest trading website in the Steam community. Worked on administration of high-profile trades & scams",
            "duration": "3 years",
            "priority": 5,
            "tags": [
                "work",
                "gaming"
            ],
            "title": "Sr. Admin, TF2Outpost"
        },
        {
            "date": [24, 'jan', 2016],
            "description": "Host based account hijack attack on php-openid",
            "priority": 6,
            "tags": [
                "security",
                "writing",
                "disclosure"
            ],
            "title": "CVE-2016-2049",
            "url": "https://cve.mitre.org/cgi-bin/cvetitle.cgi?title:CVE-2016-2049"
        },
        {
            "date": [14, 'dec', 2015],
            "description": "unique developer granted cosmetic item for the video game Team Fortress 2 granted for security issue allowing decryption of all Steam traffic",
            "priority": 6.5,
            "tags": [
                "gaming",
                "security",
                "accolade"
            ],
            "title": "Burning Flames Finder\u2019s Fee",
            "url": "http://steamcommunity.com/id/both/inventory/#440_2_4398163918"
        },
        {
            "date": [19, 'oct', 2015],
            "description": "unique developer granted cosmetic item for the video game Team Fortress 2 granted for security issues allowing remote access to computers running the video game",
            "priority": 7,
            "tags": [
                "gaming",
                "security",
                "accolade"
            ],
            "title": "Nebula Finder\u2019s Fee",
            "url": "http://steamcommunity.com/id/both/inventory/#440_2_4228772424"
        },
        {
            "date": [22, 'apr', 2014],
            "description": "unique developer granted cosmetic item for the video game Team Fortress 2 granted for security issues allowing movement millions of dollars of virtual items between arbitrary accounts via account takeover",
            "priority": 6.5,
            "tags": [
                "gaming",
                "security",
                "accolade"
            ],
            "title": "Sunbeams Ebenezer",
            "url": "http://steamcommunity.com/id/both/inventory/#440_2_4818206214"
        },
        {
            "date": [1, 'feb', 2011],
            "description": "Young Rewired State 2011",
            "priority": 5,
            "tags": [
                "software",
                "accolade"
            ],
            "title": "Best example of Coding",
            "url": "https://web.archive.org/web/20120306190316/http://youngrewiredstate.org/2011-08/cant-vote-but-can-put-a-wind-in-governments-sails/"
        },
        {
            "date": [5, 'aug', 2011],
            "description": "Rewired State: Parliament",
            "priority": 5,
            "tags": [
                "software",
                "accolade"
            ],
            "title": "Better understanding of the work of Parliament Prize",
            "url": "https://web.archive.org/web/20121105174535/http://rewiredstate.org:80/blog/2011/11/press-release-for-rewired-state-parliament"
        },
        {
            "date": [1, 'apr', 2011],
            "description": "London Real Time Hackathon",
            "priority": 5,
            "tags": [
                "software",
                "accolade"
            ],
            "title": "Geckoboard Prize",
            "url": "https://web.archive.org/web/20121113024249/http://Londonrealtime.co.uk/"
        },
        {
            "date": [1, 'apr', 2011],
            "description": "National Hack the Government Day 2011",
            "priority": 5,
            "tags": [
                "software",
                "accolade"
            ],
            "title": "Wallace and Gromit Prize",
            "url": "https://www.theguardian.com/info/developer-blog/2011/apr/05/national-hack-the-government-day-2011"
        },
        {
            "date": [1, 'jan', 2013],
            "description": "international chemistry challenge",
            "priority": 6,
            "tags": [
                "science",
                "accolade"
            ],
            "title": "7th place Cambridge Chemistry Challenge (C3L6)"
        },
        {
            "date": [1, 'jan', 2014],
            "description": "international chemistry challenge",
            "priority": 5,
            "tags": [
                "science",
                "accolade"
            ],
            "title": "5th place, Cambridge Chemistry Challenge (C3L6)"
        },
        {
            "date":[8, 'nov', 2011],
            "description": "Interview on National Hack the Government Day prize (dutch)",
            "priority": 5,
            "tags": [
                "software",
                "comment"
            ],
            "title": "MozFest: Rewired State geeft jonge programmeurs een kans",
            "url": "http://www.denieuwereporter.nl/2011/11/mozfest-rewired-state-geeft-jonge-programmeurs-een-kans/"
        },
        {
            "date": [1, 'sep', 2014],
            "description": "first security engineer at the video game streaming website. Designed security architecture for flagship projects including bits, the Twitch API, extensions and Twitch's OIDC / OAuth AuthN/Z systems. Created and defined security relationships and processes. Built Go security static analysis system, security frameworks and libraries",
            "duration": "ongoing",
            "priority": 9,
            "tags": [
                "software",
                "security",
                "work"
            ],
            "title": "Security Engineer, Twitch",
            "url": "https://twitch.tv"
        },
        {
            "date": [1, 'mar', 2012],
            "description": "charity focused on teaching code literacy. Ran and participated in hackathons for good causes. Taught software engineering to young people",
            "duration": "3 years",
            "priority": 6,
            "tags": [
                "software",
                "security",
                "work"
            ],
            "title": "Developer, Rewired State"
        }
    ],
    "who": {
        "handle": "zemnmez",
        "name": [
            "thomas",
            "neil",
            "james",
            "shadwell"
        ]
    }
})