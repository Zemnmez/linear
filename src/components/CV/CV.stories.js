
import React from 'react';

import { storiesOf } from 'lib/stories'
import { MemoryRouter as Router } from 'react-router'

import bio from 'bio'
import CV from 'components/CV'

const { timeline } = bio

storiesOf(module)
  .add('default', () => <Router><CV {...{
    data: bio,
    style: {
      "--fgc": "black",
      "--bgc": "white"
    }
  }}/></Router>)
