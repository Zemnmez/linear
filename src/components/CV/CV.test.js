import React from 'react';
import CV from "./index.js"
import renderer from 'react-test-renderer';
import bio from 'bio.js';

it('renders correctly', () => {
  const tree = renderer
    .create(<CV {...{
      data: bio
    }}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
