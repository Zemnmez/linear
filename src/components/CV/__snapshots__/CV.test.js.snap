// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="cv "
>
  <div
    className="header "
  >
    <a
      className="website"
      href=""
    >
      zemn.me
    </a>
    <div
      className="email "
    />
    <div
      className="phone "
    />
    <svg
      className="sadIcon headerIcon"
      viewBox="0 0 17.78 7.81"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        style={
          Object {
            "stroke": "var(--fgc)",
          }
        }
        transform="translate(-13.03 -62.53)"
      >
        <path
          d="M16.73 62.66l-3.47 6.02h17.32l-3.47-6.02z"
          fill="none"
          strokeWidth=".26"
        />
        <circle
          cx="21.92"
          cy="65.47"
          fill="none"
          r="1.61"
          strokeWidth=".16"
        />
        <ellipse
          cx="21.92"
          cy="65.47"
          fill="none"
          rx="3.23"
          ry="1.58"
          strokeWidth=".23"
        />
        <path
          d="M23.53 68.65a1.61 1.61 0 0 1-3.22 0c0-.9.72-1.2 1.61-1.62.9.42 1.61.73 1.61 1.62z"
          strokeWidth=".16"
          style={
            Object {
              "fill": "var(--bgc)",
            }
          }
        />
        <circle
          cx="21.92"
          cy="65.47"
          r=".54"
          strokeWidth=".08"
          style={
            Object {
              "fill": "var(--fgc)",
            }
          }
        />
      </g>
    </svg>
    <div
      className="date "
    >
      <time
        dateTime={"1970-01-01T00:00:00.008Z"}
      >
        Jan 1970
      </time>
    </div>
    <div
      className="name name"
    >
      thomas nj shadwell
    </div>
  </div>
  <div
    className="rule experienceTitle"
  >
    <span>
      selected experience
    </span>
  </div>
  <div
    className="experience "
  >
    <div
      className="work "
    >
      <div
        className="employer"
      >
        UK National Cyber Security Centre
      </div>
      <div
        className="position"
      >
        Application Security Engineer
      </div>
      <div
        className="date start"
      >
        <time
          dateTime={"2018-01-22T00:00:00.000Z"}
        >
          2018
        </time>
      </div>
      <div
        className="end"
      >
        Present
      </div>
      <div
        className="duration"
      >
        <time
          dateTime={"2018-01-01T00:00:00.000Z"}
        >
          1 year
        </time>
      </div>
      <span
        className="description content"
      >
        <p>
          advisory position. Provided expertise to UK cyber advisory / defence group on Go and building security analysis systems
        </p>
      </span>
    </div>
    <div
      className="work "
    >
      <div
        className="employer"
      >
        Twitch
      </div>
      <div
        className="position"
      >
        Security Engineer
      </div>
      <div
        className="date start"
      >
        <time
          dateTime={"2014-08-31T23:00:00.000Z"}
        >
          2014
        </time>
      </div>
      <div
        className="end"
      >
        Present
      </div>
      <div
        className="duration"
      >
        <time
          dateTime={"2014-01-01T00:00:00.000Z"}
        >
          5 years
        </time>
      </div>
      <span
        className="description content"
      >
        <p>
          first security engineer at the video game streaming website. Designed security architecture for flagship projects including bits, the Twitch API, extensions and Twitch's OIDC / OAuth AuthN/Z systems. Created and defined security relationships and processes. Built Go security static analysis system, security frameworks and libraries
        </p>
      </span>
    </div>
    <div
      className="work "
    >
      <div
        className="employer"
      >
        Consultant
      </div>
      <div
        className="position"
      >
        Software Engineer
      </div>
      <div
        className="date start"
      >
        <time
          dateTime={"2012-04-30T23:00:00.000Z"}
        >
          2012
        </time>
      </div>
      <div
        className="date end"
      >
        <time
          dateTime={"2014-04-30T23:00:00.000Z"}
        >
          2014
        </time>
      </div>
      <div
        className="duration"
      >
        <time
          dateTime={"2012-04-30T23:00:00.000Z"}
        >
          2 years
        </time>
      </div>
      <span
        className="description content"
      >
        <p>
          full stack freelance work building MVPs for London startups and wrangling data for hackathons
        </p>
      </span>
    </div>
    <div
      className="work "
    >
      <div
        className="employer"
      >
        Rewired State
      </div>
      <div
        className="position"
      >
        Developer
      </div>
      <div
        className="date start"
      >
        <time
          dateTime={"2012-03-01T00:00:00.000Z"}
        >
          2012
        </time>
      </div>
      <div
        className="date end"
      >
        <time
          dateTime={"2015-03-01T00:00:00.000Z"}
        >
          2015
        </time>
      </div>
      <div
        className="duration"
      >
        <time
          dateTime={"2012-03-01T00:00:00.000Z"}
        >
          3 years
        </time>
      </div>
      <span
        className="description content"
      >
        <p>
          charity focused on teaching code literacy. Ran and participated in hackathons for good causes. Taught software engineering to young people
        </p>
      </span>
    </div>
  </div>
  <div
    className="rule worksTitle"
  >
    <span>
      of note
    </span>
  </div>
  <div
    className="skills "
  >
    <div
      className="title"
    >
      AWS
    </div>
    <div
      className="description"
    />
    <div
      className="title"
    >
      Bash
    </div>
    <div
      className="description"
    />
    <div
      className="title"
    >
      Go
    </div>
    <div
      className="description"
    />
    <div
      className="title"
    >
      Javascript
    </div>
    <div
      className="description"
    />
    <div
      className="title"
    >
      Python
    </div>
    <div
      className="description"
    />
    <div
      className="title"
    >
      React
    </div>
    <div
      className="description"
    />
    <div
      className="title"
    >
      Ruby
    </div>
    <div
      className="description"
    />
    <div
      className="title"
    >
      appsec arch
    </div>
    <div
      className="description"
    />
    <div
      className="title"
    >
      code review
    </div>
    <div
      className="description"
    />
    <div
      className="title"
    >
      cryptography
    </div>
    <div
      className="description"
    />
    <div
      className="title"
    >
      d3
    </div>
    <div
      className="description"
    />
    <div
      className="title"
    >
      electron
    </div>
    <div
      className="description"
    />
  </div>
  <div
    className="rule skillsTitle"
  >
    <span>
      skills
    </span>
  </div>
  <div
    className="skills works"
  >
    <div
      className="title"
    >
      when security creates insecurity
    </div>
    <div
      className="description"
    >
      exploit using content security policy 1 to steal data on the web
    </div>
    <div
      className="title"
    >
      Steam Remote Code Execution
    </div>
    <div
      className="description"
    >
      vulnerability to remotely access Steam users' computers
    </div>
    <div
      className="title"
    >
      forbes 30 under 30, tech
    </div>
    <div
      className="description"
    >
      for my work at Twitch, and on responsible disclosure
    </div>
    <div
      className="title"
    >
      how to hack the uk tax system, i guess
    </div>
    <div
      className="description"
    >
      vulnerability allowing manipulation of UK tax system
    </div>
  </div>
  <div
    className="skills works2"
  >
    <div
      className="title"
    >
      $7,500 Steam Weakness Let Hackers Take Remote Control Of Gamers' PCs
    </div>
    <div
      className="description"
    >
      news coverage of steam rce
    </div>
    <div
      className="title"
    >
      Ãœbersicht Remote Code Execution, Spotify takeover
    </div>
    <div
      className="description"
    >
      Quick article on the security of modern desktop web applications
    </div>
    <div
      className="title"
    >
      'Mr. Robot' Web Weaknesses Left Fans And USA Network Vulnerable, Warns Non-Fictional Hacker
    </div>
    <div
      className="description"
    >
      code execution in official Mr Robot site
    </div>
    <div
      className="title"
    >
      steam patches broken crypto in wake of replay, padding oracle attacks
    </div>
    <div
      className="description"
    >
      padding oracle based decryption of Steam traffic
    </div>
  </div>
  <svg
    className="future future"
    height="348"
    version="1"
    viewBox="0 0 446 348"
    width="446"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M174 0L54 120l33 32L207 33 174 0zm98 0l-32 33 119 119 33-32L272 0zm-49 59L109 174l114 114 115-114L223 59zM33 141L0 174l33 33 33-33-33-33zm380 0l-32 33 32 33 33-33-33-33zM87 195l-33 33 120 120 33-33L87 195zm272 0L240 315l32 33 120-120-33-33z"
      vectorEffect="non-scaling-stroke"
    />
  </svg>
</div>
`;
